### PR TITLE
document build.js

### PR DESCRIPTION
**Pull request by ghost**
this is something that i've been wanting to do for a longer time now. let's document the quite complicated (for a metalsmith build script at least) `build.js` file.

i've started and added some base comments. if any of those are wrong or superfluous, please let me know. also let me know if there's something left to be documented!

### CODE REVIEW

**Comment by ghost** [comment]
ping @nodejs/website

**Comment by Fishrock123** [comment]
I think most of that was written between @fhemberger and @phillipj so they are probably best to review.

**Comment by stevemao** [comment]
:+1: for comments!!!

File: build.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
-#!/usr/bin/env node
+#! /usr/bin/env node
 
 'use strict'
 
+// BUILD.JS: This file is responsible for building the HTML site and serving it
+// as a static server.
```
Comment (phillipj) [inline build.js]: "This file is responsible for building static HTML pages and a server for local development."?

File: build.js [hunk 2]
```diff
@@ -17,26 +20,30 @@ const fs = require('fs')
 const ncp = require('ncp')
 const junk = require('junk')
 
+// Require custom scripts that we'll need for the build process.
```
Comment (phillipj) [inline build.js]: Not sure how informative this comment is

File: build.js [hunk 3]
```diff
@@ -162,14 +184,20 @@ function buildlocale (source, locale) {
         apidocslink: require('./scripts/helpers/apidocslink.js')
       }
     }))
+    // Pipes the generated files into their respective subdirectory in the build
+    // directory.
     .destination(path.join(__dirname, 'build', locale))
 
+  // This actually executes the build and stops the internal timer after
+  // completition.
```
Comment (phillipj) [inline build.js]: completion

File: build.js [hunk 4]
```diff
@@ -210,8 +243,9 @@ function fullbuild () {
   })
 }
 
+// The server function, where the site is exposed through a static file server.
```
Comment (phillipj) [inline build.js]: The server function, where the site is exposed through a static file server **locally**.

**Comment by phillipj** [comment]
Very nice work @sup! The setup isn't very self explanatory for new comers, hopefully these comments will make it a lot easier too grasp :+1: 

Other than the comments, this LGTM.

**Pull request merged**